surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,361 @@
+debugging not all the systems you target.
+
+Type: select all that apply (optional)
+
+- Linux
+- Windows
+- Windows Subsystem for Linux
+- Mac OS
+- Embedded devices
+- Other (open response)
+
+> *Justification:* Cohort analysis
+>
+> Debuggers are often platform-specific and various improvements that the
+> compiler team could make could only benefit a subset of all platforms, so
+> knowing what users are using is valuable.
+
+### How would you rate your Rust expertise?
+Type: select one
+
+- I can't read or write Rust
+- I can write simple programs in Rust
+- I can write useful, production-ready code but it is a struggle
+- I am productive writing Rust
+- I am an expert
+
+> *Justification:* Cohort analysis
+>
+> Users with different levels of experience are likely to face different
+> challenges.
+
+
+### What forms of debugging do you typically use with Rust?
+Type: select all that apply
+
+- `println!`/`log`/`tracing`/`dbg!`
+- Stack traces (printed by `panic!()`/`backtrace-rs`)
+- Traditional stepping debuggers (gdb/lldb/WinDbg/Visual Studio)
+- Record/replay debuggers (rr/WinDbg or Visual Studio with Time Travel Debugging)
+- High-level tracing (`dtrace`/`ebpf`/`ETW`)
+
+> *Justification:*
+>
+> Understand how users are debugging Rust code.
+
+### What uses of Rust are you debugging?
+Type: select all that apply (optional)
+
+- Single-threaded application
+- Multi-threaded application
+- Rust applications using `async`/`await`
+- Pure Rust application
+- Mixed Rust and other language application
+- Crash dumps
+- Cloud/server-side application
+- Embedded hardware
+- WebAssembly
+
+> *Justification:*
+>
+> Some debugger use cases are very different from "standard" debugging.
+
+### Do you debug Rust remotely?
+i.e. are you debugging something that's running on an embedded device/another
+platform, or something like that
+
+Type: select all that apply
+
+- Yes, debugger on the host (e.g. `gdbserver` or wired connection to embedded device)
+- Yes, debugger on the device (e.g. an SSH session with a debugger running)
+- No
+
+> *Justification:*
+>
+> Remote debugging will have a different user experience that might be better
+> or worse.
+
+### When do you start debugging Rust?
+Type: select all that apply
+
+- Fixing a crash/panic/access violation
+- Fixing a logic bug
+- Fixing a misbehaving program
+- Investigating a crash dump
+- Learning how a well-behaving program works
+- Profiling
+- Understand a third-party library's behaviour
+
+> *Justification:*
+>
+> Different motivations for debugging will benefit from different features from
+> debuggers and might have a better or worse experience.
+
+
+### How would you rate your expertise with debuggers?
+Type: select one
+
+- I have never used a debugger
+- I am uncomfortable using a debugger [`NEXT`](#why-are-you-uncomfortable-using-a-debugger)
+- I am comfortable using a debugger
+- I am very comfortable using a debugger
```

Comment:
This is not a very clear answer. I would just drop it, but if you want to keep it, maybe change the wording and/or give examples

Reply:
I've removed the "very comfortable" option because I think the difference between "very comfortable" and "comfortable" is a vague but the others are clearer.